### ISSUE TITLE
Calibration upload 585d4d

### ISSUE THREAD

**Issue by bronger** [lensfun/lensfun#299]
Calibration images were uploaded to the directory that starts with “`585d4d_`”.

Please read the [workflow description](https://github.com/lensfun/lensfun/blob/master/tools/calibration_webserver/workflow.rst) for further instructions about the calibration.

Additional comments by the uploader:

> PC-Nikkor 35mm f/2.8. All shots at f8. Multiple levels of shift, noted in file name. 
> 
>  If the shots aren't good enough I'll shoot them again. I was tilting the camera down as I was shifting the lens up, to keep the lines in the right position.
> 
> I took shots at a number of different levels of shift. I don't know how much trouble it is to create the correction profiles, but if you have the time it would be great to have profiles covering the full range of shift, as the distortion seems to increase or change with the shift. I shot 2 sets. 0-2-4-6-8-10-11 & and a more condensed 0-3-6-9-11
> 
> This lens is fantastic on one hand, but also has very pronounced distortion which interferes with architectural shots. 
> 
> Thanks for your time and efforts! looking forward to the profiles!

**Comment by junkyardsparkle** [comment]
@uploader I haven't had a chance to look at the shots yet, but I don't think lensfun has a correction model that can handle shifted lens distortion. I think you would have to somehow "correct" the shift first, apply the lens correction, then "uncorrect" the shift... but maybe somebody smarter than me knows better.

**Comment by PhilBaz** [comment]
Ahh, understandable if Lensfun doesn't have that ability. Could a different profile be developed for each level of shift? as its a prime lens I figured that perhaps there could be 4 or 5 correction profiles (0mm-3mm-6mm-9mm-11mm) rather than one that accounted for shift amount. Would that work?

 I would love to get the distortion on this lens corrected as its quite noticeable. If you have any ideas id be happy to hear them. Thanks. Phil

**Comment by junkyardsparkle** [comment]
The problem is that lensfun assumes circular symmetry for the distortion, which is only the case *before* the shift... you can see, for instance, in the last shot how the distortion is "barrel" in the top half of the image, but "pincushion" in the lower half. We can certainly create a correction profile for the zero-shift image, though - the uploaded file looks good.

It might be possible to hack together some workflow that does the unshift>undistort>reshift process, but it would involve multiple pixel interpolation steps, and I suspect the shift modelling would have to be pretty accurate for good results, particularly with respect to TCA... but there are probably people who would know better than I do. :)

**Comment by junkyardsparkle** [comment]
@PhilBaz Incidentally, [it sounds like there are two distinct versions of this lens](http://mir.com.my/rb/photography/companies/nikon/nikkoresources/PC_Nikkor/index2.htm). Would this be the post-1980 "black knob" version?

**Comment by PhilBaz** [comment]
Circular symmetry, that makes perfect sense. 

Hummm, i remember reading about applying distortion correction to the rokinon 24mm shift lens. I guess that's similar to the process you are saying might be possible. 

Essentially you crop the photo (expanding, rather than contracting the crop) so that the image is in the correct part of the total image circle the lens is actually capable of, apply the correction, essentially correcting the entire image circle with the photo in the corresponding section of the total image/circle, then crop it back down.  (hope that makes sense)  is that what you were thinking?

**Comment by PhilBaz** [comment]
Ahhh, yes later 'black knob' model.... joke repressed.

**Comment by junkyardsparkle** [comment]
Something like that, but it might have to be more sophisticated than that. The basic functionality might be present in panotools, but I'm not sure interfaces like hugin etc are set up to apply it in the right order. It's not something I've tried to get my head around yet, but other people probably have... @bronger could probably tell you. In any case, the profile for the basic zero-shift case has been added to the database. Nice job getting a shot with the straight lines for distortion and extra contrasty detail for tca correction all-in-one!
